ada/testsuite/ada/nameres.adb

```diff
@@ -291,9 +291,19 @@ procedure Nameres is
                if not Quiet then
                   Put_Line ("Expr: " & N.Short_Image);
                   if Kind (N) in Ada_Name then
-                     Put_Line
-                       ("  references: " & Image (P_Xref
-                         (As_Name (N), Args.Imprecise_Fallback.Get)));
+                     declare
+                        Referenced_Decl_Image : String :=
+                           (if Show_Slocs
+                            then Image
+                              (P_Xref (As_Name (N),
+                                       Args.Imprecise_Fallback.Get))
+                            else Image
+                              (P_Unique_Identifying_Name (P_Basic_Decl (P_Xref
+                                (As_Name (N), Args.Imprecise_Fallback.Get)))));
```

Comment:
There is opportunity to factor common subexpression (call to P_Xref). Would make the whole expression more readable IMO.